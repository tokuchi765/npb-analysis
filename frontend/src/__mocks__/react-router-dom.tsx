import React from 'react';
const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: any) => <div>{children}</div>;
module.exports = rrd;
