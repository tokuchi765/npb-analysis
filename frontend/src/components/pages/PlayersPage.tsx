import React, { useEffect, useState } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { useHistory } from 'react-router-dom';
import GenericTemplate from '../templates/GenericTemplate';
import TablePages, { HeadCell } from './TablePages';
import axios from 'axios';
import _ from 'lodash';
import * as H from 'history';

const teamNameList = [
  'Giants',
  'Baystars',
  'Tigers',
  'Carp',
  'Dragons',
  'Swallows',
  'Lions',
  'Hawks',
  'Eagles',
  'Marines',
  'Fighters',
  'Buffaloes',
];

interface PlayerDate {
  main: string;
  position: string;
  height: string;
  draft: string;
  career: string;
}
const headCells: HeadCell[] = [
  { id: 'main', numeric: false, disablePadding: true, label: '選手名' },
  { id: 'position', numeric: false, disablePadding: true, label: 'ポジション' },
  { id: 'height', numeric: false, disablePadding: true, label: '身長' },
  { id: 'draft', numeric: false, disablePadding: true, label: 'ドラフト' },
  { id: 'career', numeric: false, disablePadding: true, label: '経歴' },
];

function createPlayerDate(
  main: string,
  position: string,
  height: string,
  draft: string,
  career: string
) {
  const result: PlayerDate = { main, position, height, draft, career };
  return result;
}

function createPlayerDates(
  careers: {
    Name: string;
    Position: string;
    Height: string;
    Draft: string;
    Career: string;
  }[]
) {
  const playerDateList: PlayerDate[] = [];
  careers.forEach((career) => {
    playerDateList.push(
      createPlayerDate(career.Name, career.Position, career.Height, career.Draft, career.Career)
    );
  });
  return playerDateList;
}

function createPlayerIds(
  careers: {
    PlayerID: string;
    Name: string;
  }[]
) {
  const playerIdMap: Map<string, string> = new Map<string, string>();
  careers.forEach((career) => {
    playerIdMap.set(career.Name, career.PlayerID);
  });
  return playerIdMap;
}

function getTeamId(teamName: string) {
  for (const index in teamNameList) {
    if (teamNameList[index] === teamName) {
      return _.padStart(String(Number(index) + 1), 2, '0');
    }
  }
}

interface PageProps extends RouteComponentProps<{ id: string }> {
  history: H.History;
  location: H.Location<any>;
}

const PlayersPage: React.FC<PageProps> = (props) => {
  const [playerDates, setPlayerDates] = useState<PlayerDate[]>([]);
  const [playerIdMap, setPlayerIds] = useState<Map<string, string>>(new Map<string, string>());
  const [initTeam, setInitTeam] = useState<string>('');
  const history = useHistory();

  const getPlayerList = async (teamName: string) => {
    const teamID = getTeamId(teamName);
    const result = await axios.get(`http://localhost:8081/team/careers/${teamID}/2020`);
    setPlayerIds(createPlayerIds(result.data.careers));
    setPlayerDates(createPlayerDates(result.data.careers));
    setInitTeam(teamName);

    history.push({
      state: { teamName: teamName },
    });
  };

  const getTeamName = (location: any) => {
    let teamName = location.state && location.state.teamName;
    if (teamName === undefined) {
      teamName = 'Hawks';
    }
    return teamName;
  };

  useEffect(() => {
    (async () => {
      const teamName = getTeamName(props.location);
      getPlayerList(teamName);
    })();
  }, []);

  return (
    <GenericTemplate title="選手一覧ページ">
      <TablePages
        title={'選手一覧'}
        setSelect={setInitTeam}
        getDataList={getPlayerList}
        datas={playerDates}
        selects={teamNameList}
        headCells={headCells}
        initSorted={'main'}
        initSelect={initTeam}
        selectLabel={'チーム'}
        mainLink={true}
        linkValues={playerIdMap}
        path={'/player/'}
      />
    </GenericTemplate>
  );
};

export default PlayersPage;
