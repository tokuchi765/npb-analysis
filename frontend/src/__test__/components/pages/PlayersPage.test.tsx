import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import PlayersPage from '../../../components/pages/PlayersPage';
import { MemoryRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

describe('選手一覧ページテスト', () => {
  it('スナップショット作成', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <PlayersPage history={{} as any} location={{} as any} match={{} as any} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
