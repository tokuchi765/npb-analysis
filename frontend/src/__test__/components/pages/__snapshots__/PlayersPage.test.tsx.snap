// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`選手一覧ページテスト スナップショット作成 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary makeStyles-appBar-4 makeStyles-appBarShift-5 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 makeStyles-menuButtonHidden-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h1
        className="MuiTypography-root makeStyles-title-8 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        管理画面
      </h1>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-10 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="makeStyles-toolbarIcon-3"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <a
          className="makeStyles-link-16"
          href="/"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                トップページ
              </span>
            </div>
          </div>
        </a>
        <li
          className="MuiListItem-root makeStyles-link-16 MuiListItem-gutters"
          disabled={false}
          onMouseOver={[Function]}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GroupsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              チーム情報
            </span>
          </div>
        </li>
        <a
          className="makeStyles-link-16"
          href="/players"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                選手一覧ページ
              </span>
            </div>
          </div>
        </a>
        <a
          className="makeStyles-link-16"
          href="/manager"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                監督ページ
              </span>
            </div>
          </div>
        </a>
        <a
          className="makeStyles-link-16"
          href="/search"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                選手検索ページ
              </span>
            </div>
          </div>
        </a>
      </ul>
    </div>
  </div>
  <main
    className="makeStyles-content-13"
  >
    <div
      className="makeStyles-appBarSpacer-12"
    />
    <div
      className="MuiContainer-root makeStyles-container-14 MuiContainer-maxWidthLg"
    >
      <h2
        className="MuiTypography-root makeStyles-pageTitle-9 MuiTypography-h5 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        選手一覧ページ
      </h2>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1ps6pg7-MuiPaper-root"
      >
        <div
          className="MuiTypography-root makeStyles-title-19 MuiTypography-h6"
          id="tableTitle"
        >
          <div
            className="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row makeStyles-grid-22 css-1unhvdk-MuiGrid2-root"
          >
            <div
              className="MuiGrid2-root MuiGrid2-direction-xs-row css-27y2fm-MuiGrid2-root"
            >
              <div
                className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-21 css-1ps6pg7-MuiPaper-root"
              >
                _
                _
                選手一覧
              </div>
            </div>
            <div
              className="MuiGrid2-root MuiGrid2-direction-xs-row css-27y2fm-MuiGrid2-root"
            >
              <div
                className="MuiFormControl-root makeStyles-formControl-20"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  id="demo-simple-select-label"
                >
                  チーム
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="demo-simple-select-label demo-simple-select"
                    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="demo-simple-select"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    aria-hidden={true}
                    className="MuiSelect-nativeInput"
                    onAnimationStart={[Function]}
                    onChange={[Function]}
                    required={false}
                    tabIndex={-1}
                    value=""
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiFormControl-root makeStyles-formControl-20"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  id="demo-simple-select-label"
                >
                  年
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="demo-simple-select-label demo-simple-select"
                    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="demo-simple-select"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    aria-hidden={true}
                    className="MuiSelect-nativeInput"
                    onAnimationStart={[Function]}
                    onChange={[Function]}
                    required={false}
                    tabIndex={-1}
                    value=""
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <table
          aria-label="simple table"
          className="MuiTable-root makeStyles-table-17"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort="descending"
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  選手名
                  <span
                    className="makeStyles-visuallyHidden-18"
                  >
                    sorted descending
                  </span>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  ポジション
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  身長
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  ドラフト
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
                scope="col"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  経歴
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiBox-root MuiBox-root-23"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Copyright © 
          <a
            color="inherit"
            href="/"
            onClick={[Function]}
          >
            管理画面
          </a>
           
          2023
          .
        </p>
      </div>
    </div>
  </main>
</div>
`;
